tiledb/sm/serialization/array_schema.cc

```diff
@@ -304,110 +304,139 @@ Status dimension_to_capnp(
   return Status::Ok();
 }
 
-Status dimension_from_capnp(
-    const capnp::Dimension::Reader& dimension_reader,
-    tdb_unique_ptr<Dimension>* dimension) {
-  Datatype dim_type = Datatype::ANY;
-  RETURN_NOT_OK(datatype_enum(dimension_reader.getType().cStr(), &dim_type));
-  dimension->reset(tdb_new(Dimension, dimension_reader.getName(), dim_type));
+tuple<Status, optional<ByteVecValue>> tile_extent_from_capnp(
```

Comment:
There's no particularly good reason for this function to return a `Status`, since this code was written after we have exception safety.